# diff --git a/app/CMakeLists.txt b/app/CMakeLists.txt
# index dcf486f..6e31d59 100644
# --- a/app/CMakeLists.txt
# +++ b/app/CMakeLists.txt
# @@ -84,6 +84,22 @@ if (KF5Kipi_FOUND)
#      target_link_libraries(gwenview KF5::Kipi)
#  endif()
#  
# +if (APPLE)
# +    # own plist template
# +    set_target_properties (gwenview PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/MacOSXBundleInfo.plist.in)
# +    # the MacOSX bundle display name property (CFBundleDisplayName) is not currently supported by cmake,
# +    # so has to be set for all targets in this cmake file
# +    set(MACOSX_BUNDLE_DISPLAY_NAME Gwenview5)
# +    set_target_properties(gwenview PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.gwenview")
# +    set_target_properties(gwenview PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Gwenview5")
# +    set_target_properties(gwenview PROPERTIES MACOSX_BUNDLE_DISPLAY_NAME "Gwenview5")
# +    set_target_properties(gwenview PROPERTIES MACOSX_BUNDLE_INFO_STRING "Gwenview, KDE Image Viewer")
# +    set_target_properties(gwenview PROPERTIES MACOSX_BUNDLE_LONG_VERSION_STRING "Gwenview ${KDE_APPLICATIONS_VERSION}")
# +    set_target_properties(gwenview PROPERTIES MACOSX_BUNDLE_SHORT_VERSION_STRING "${KDE_APPLICATIONS_VERSION_MAJOR}.${KDE_APPLICATIONS_VERSION_MINOR}")
# +    set_target_properties(gwenview PROPERTIES MACOSX_BUNDLE_BUNDLE_VERSION "${KDE_APPLICATIONS_VERSION}")
# +    set_target_properties(gwenview PROPERTIES MACOSX_BUNDLE_COPYRIGHT "2002-2016 The Gwenview Authors")
# +endif (APPLE)
# +
#  install(TARGETS gwenview
#      ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
#  
diff --git a/app/MacOSXBundleInfo.plist.in b/app/MacOSXBundleInfo.plist.in
new file mode 100644
index 0000000..b0622c6
--- /dev/null
+++ b/app/MacOSXBundleInfo.plist.in
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+        <key>NSPrincipalClass</key>
+        <string>NSApplication</string>
+        <key>NSHighResolutionCapable</key>
+        <string>True</string>
+        <key>CFBundleDevelopmentRegion</key>
+        <string>English</string>
+        <key>CFBundleExecutable</key>
+        <string>${MACOSX_BUNDLE_EXECUTABLE_NAME}</string>
+        <key>CFBundleGetInfoString</key>
+        <string>${MACOSX_BUNDLE_INFO_STRING}</string>
+        <key>CFBundleIconFile</key>
+        <string>${MACOSX_BUNDLE_ICON_FILE}</string>
+        <key>CFBundleIdentifier</key>
+        <string>${MACOSX_BUNDLE_GUI_IDENTIFIER}</string>
+        <key>CFBundleInfoDictionaryVersion</key>
+        <string>6.0</string>
+        <key>CFBundleLongVersionString</key>
+        <string>${MACOSX_BUNDLE_LONG_VERSION_STRING}</string>
+        <key>CFBundleName</key>
+        <string>${MACOSX_BUNDLE_BUNDLE_NAME}</string>
+        <key>CFBundlePackageType</key>
+        <string>APPL</string>
+        <key>CFBundleShortVersionString</key>
+        <string>${MACOSX_BUNDLE_SHORT_VERSION_STRING}</string>
+        <key>CFBundleSignature</key>
+        <string>????</string>
+        <key>CFBundleVersion</key>
+        <string>${MACOSX_BUNDLE_BUNDLE_VERSION}</string>
+        <key>CSResourcesFileMapped</key>
+        <true/>
+        <key>LSRequiresCarbon</key>
+        <true/>
+        <key>NSHumanReadableCopyright</key>
+        <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+        <key>LSMultipleInstancesProhibited</key>
+        <true/>
+        <key>CFBundleDocumentTypes</key>
+        <array>
+            <dict>
+                <key>CFBundleTypeExtensions</key>
+                <array>
+                        <string>*</string>
+                </array>
+                <key>CFBundleTypeName</key>
+                <string>NSStringPboardType</string>
+                <key>CFBundleTypeRole</key>
+                <string>Viewer</string>
+            </dict>
+        </array>
+</dict>
+</plist>
# diff --git a/app/mainwindow.cpp b/app/mainwindow.cpp
# index 7b30c4e..b5f2b61 100644
# --- a/app/mainwindow.cpp
# +++ b/app/mainwindow.cpp
# @@ -32,6 +32,9 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#  #include <QVBoxLayout>
#  #include <QMenuBar>
#  #include <QUrl>
# +#ifdef Q_OS_OSX
# +#include <QFileOpenEvent>
# +#endif
#  
#  // KDE
#  #include <KIO/NetAccess>
# @@ -794,10 +797,16 @@ MainWindow::MainWindow()
#      d->mKIPIExportAction->setKIPIInterface(d->mKIPIInterface);
#  #endif
#      setAutoSaveSettings();
# +#ifdef Q_OS_OSX
# +    qApp->installEventFilter(this);
# +#endif
#  }
#  
#  MainWindow::~MainWindow()
#  {
# +#ifdef Q_OS_OSX
# +    qApp->removeEventFilter(this);
# +#endif
#      if (GwenviewConfig::deleteThumbnailCacheOnExit()) {
#          QDir dir(ThumbnailProvider::thumbnailBaseDir());
#          if (dir.exists()) {
# @@ -1542,6 +1551,21 @@ void MainWindow::resizeEvent(QResizeEvent* event)
#      }
#  }
#  
# +bool MainWindow::eventFilter(QObject *obj, QEvent *event)
# +{
# +#ifdef Q_OS_OSX
# +    /**
# +    * handle Mac OS X file open events (only exist on OS X)
# +    */
# +    if (event->type() == QEvent::FileOpen) {
# +        QFileOpenEvent *fileOpenEvent = static_cast<QFileOpenEvent*>(event);
# +        openUrl(fileOpenEvent->url());
# +        return true;
# +    }
# +#endif
# +    return false;
# +}
# +
#  void MainWindow::setDistractionFreeMode(bool value)
#  {
#      d->mFullScreenContent->setDistractionFreeMode(value);
# diff --git a/app/mainwindow.h b/app/mainwindow.h
# index 6d252f9..b0c77b6 100644
# --- a/app/mainwindow.h
# +++ b/app/mainwindow.h
# @@ -55,6 +55,8 @@ public:
#  
#      void setDistractionFreeMode(bool);
#  
# +    bool eventFilter(QObject *, QEvent *);
# +
#  public Q_SLOTS:
#      void showStartMainPage();
#  
